### PR TITLE
Update Wasmtime to v0.30

### PR DESCRIPTION
**Pull request by radu-matei**
This PR updates Wasmtime to v0.30 -- see https://rustsec.org/advisories/RUSTSEC-2021-0110.html

### CODE REVIEW

File: crates/wasi-provider/Cargo.toml [hunk 1]
```diff
@@ -36,12 +36,12 @@ serde_json = "1.0"
...
-wasi-common = "0.28"
-wasmtime = "0.28"
-wasmtime-wasi = "0.28"
+wasi-cap-std-sync = "0.30"
+wasi-common = "0.30"
+wasmtime = "0.30"
+wasmtime-wasi = "0.30"
 wat = "1.0.38"
-wasi-experimental-http-wasmtime = "0.5.0"
+wasi-experimental-http-wasmtime = { git = "https://github.com/radu-matei/wasi-experimental-http", branch = "wasmtime-v030" }
```
Comment (radu-matei) [inline crates/wasi-provider/Cargo.toml:44]: TODO: update once a new release is available.

**Comment by radu-matei** [comment]
Note: the failed check is because of an advisory not introduced by this PR -- see https://rustsec.org/advisories/RUSTSEC-2021-0093

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
This looks good to go once the experimental http library is updated. However, one of the dependencies in here (possibly in the lock file?) is pulling in a version of crossbeam with a security advisory. The latest patch release (0.8.1) should solve that. If it turns out to be an upstream problem, let's address the issue there

**Comment by thomastaylor312** [comment]
@radu-matei This advisory is not appearing in previous runs, so something happened here to bring in a new library. So we should address it here (or at least diagnose and open an upstream issue)

**Comment by radu-matei** [comment]
Directly updated `crossbeam-deque` to v0.8.1 - see cf7658f210f074092e0ef140a2ae095ea461ca96.

**Review by thomastaylor312 — APPROVED** [review]

**Comment by radu-matei** [comment]
@thomastaylor312 -- what do you think we should do about the license check for `webpki-roots`?

```
error[L001]: failed to satisfy license requirements
  ┌─ webpki-roots 0.21.1 (registry+rust-lang/crates.io-index):4:12
  │
4 │ license = "MPL-2.0"
  │            ^^^^^^^
  │            │
  │            license expression retrieved via Cargo.toml `license`
  │            rejected: license is considered copyleft
  │
  = webpki-roots v0.21.1
    └── reqwest v0.11.4
        ├── (dev) krustlet v1.0.0-alpha.1
        ├── (dev) kubelet v1.0.0-alpha.1
        │   ├── krustlet v1.0.0-alpha.1 (*)
        │   └── wasi-provider v1.0.0-alpha.1
        │       └── krustlet v1.0.0-alpha.1 (*)
        ├── kubelet v1.0.0-alpha.1 (*)
        ├── oci-distribution v0.7.0
        │   ├── krustlet v1.0.0-alpha.1 (*)
        │   ├── kubelet v1.0.0-alpha.1 (*)
        │   └── (dev) wasi-provider v1.0.0-alpha.1 (*)
        └── wasi-experimental-http-wasmtime v0.6.0
            └── wasi-provider v1.0.0-alpha.1 (*)
```

**Comment by thomastaylor312** [comment]
That is a known issue @radu-matei. We have an open issue with the CNCF to grant a license exception to it. For now we are leaving it failing. You should be good to merge whenever

**Pull request merged**